src/components/application_manager/test/commands/hmi/get_urls_test.cc

```diff
@@ -81,237 +81,255 @@ const std::string kInitialService = "0x0";
-      kInitialService;
 
-  ON_CALL(mock_app_manager_, event_dispatcher())
-      .WillByDefault(ReturnRef(mock_event_dispatcher_));
+#ifdef EXTENDED_POLICY
+TEST_F(GetUrlsTest, ProcessPolicyServiceURLs_SUCCESS) {
+  (*command_msg_)[am::strings::msg_params][am::hmi_request::service] =
+      kPolicyService;
 
-  RequestFromHMIPtr command(CreateCommand<GetUrls>(command_msg));
+  EXPECT_CALL(mock_policy_handler_, PolicyEnabled()).WillOnce(Return(true));
 
-  EXPECT_CALL(mock_app_manager_, GetPolicyHandler());
+  EndpointUrls endpoints_;
+  EndpointData data(kDefaultUrl);
+  endpoints_.push_back(data);
 
-  EXPECT_CALL(policy_handler_, PolicyEnabled()).WillOnce(Return(false));
+  ON_CALL(mock_policy_handler_, GetServiceUrls(kPolicyService, _))
+      .WillByDefault(SetArgReferee<1>(endpoints_));
+
+  MockAppPtr mock_app = CreateMockApp();
 
-  EXPECT_CALL(mock_app_manager_, ManageHMICommand(command_msg))
+  EXPECT_CALL(mock_policy_handler_, GetAppIdForSending())
+      .WillOnce(Return(kAppIdForSending));
+
+  EXPECT_CALL(mock_app_manager_, application(kAppIdForSending))
+      .WillOnce(Return(mock_app));
+  EXPECT_CALL(*mock_app, app_id()).WillOnce(Return(kAppIdForSending));
+  EXPECT_CALL(mock_app_manager_, ManageHMICommand(command_msg_))
       .WillOnce(Return(true));
 
-  command->Run();
+  request_command_->Run();
 
-  EXPECT_EQ((*command_msg)[strings::params][strings::message_type].asInt(),
+  EXPECT_FALSE((*command_msg_)[am::strings::msg_params].keyExists(
+      am::hmi_request::service));
+
+  EXPECT_EQ((*command_msg_)[strings::params][strings::message_type].asInt(),
             am::MessageType::kResponse);
-  EXPECT_EQ((*command_msg)[strings::params][am::hmi_response::code].asInt(),
-            Common_Result::DATA_NOT_AVAILABLE);
+  EXPECT_EQ((*command_msg_)[strings::params][am::hmi_response::code].asInt(),
+            Common_Result::SUCCESS);
+
+  EXPECT_EQ(kAppIdForSending,
+            (*command_msg_)[am::strings::msg_params][am::hmi_response::urls][0]
+                           [strings::app_id].asInt());
+  EXPECT_EQ(kDefaultUrl,
+            (*command_msg_)[am::strings::msg_params][am::hmi_response::urls][0]
+                           [strings::url].asString());
 }
-TEST_F(GetUrlsTest, RUN_EmptyEndpoints_UNSUCCESS) {
-  MessageSharedPtr command_msg(
-      CreateMessage(NsSmartDeviceLink::NsSmartObjects::SmartType_Map));
-  (*command_msg)[am::strings::msg_params][am::strings::number] = "123";
-  (*command_msg)[am::strings::params][am::strings::connection_key] =
-      kConnectionKey;
-  (*command_msg)[am::strings::msg_params][am::hmi_request::service] =
-      kInitialService;
 
-  ON_CALL(mock_app_manager_, event_dispatcher())
-      .WillByDefault(ReturnRef(mock_event_dispatcher_));
+TEST_F(GetUrlsTest, ProcessPolicyServiceURLs_IncorrectIdForSending_UNSUCCESS) {
+  (*command_msg_)[am::strings::msg_params][am::hmi_request::service] =
+      kPolicyService;
```

Comment:
@LevchenkoS Well the "am::hmi_request::service" field is present in fixture constructor and has default value "kInitialService" which is used by majority of the unit cases.